core/src/test/java/com/devonfw/tools/solicitor/componentinfo/scancode/UncuratedScancodeComponentInfoProviderTests.java

```diff
@@ -0,0 +1,98 @@
+package com.devonfw.tools.solicitor.componentinfo.scancode;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import com.devonfw.tools.solicitor.common.content.web.DirectUrlWebContentProvider;
+import com.devonfw.tools.solicitor.common.packageurl.AllKindsPackageURLHandler;
+import com.devonfw.tools.solicitor.componentinfo.ComponentInfoAdapterException;
+import com.devonfw.tools.solicitor.componentinfo.curation.SingleFileCurationProvider;
+
+/**
+ * This class contains JUnit test methods for the {@link UncuratedScancodeComponentInfoProvider} class.
+ */
+public class UncuratedScancodeComponentInfoProviderTests {
+
+  // the object under test
+  UncuratedScancodeComponentInfoProvider uncuratedScancodeComponentInfoProvider;
+
+  FileScancodeRawComponentInfoProvider fileScancodeRawComponentInfoProvider;
+  
+  SingleFileCurationProvider singleFileCurationProvider;
+  
+  @BeforeEach
+  public void setup() {
+
+    AllKindsPackageURLHandler packageURLHandler = Mockito.mock(AllKindsPackageURLHandler.class);
+
+    Mockito.when(packageURLHandler.pathFor("pkg:maven/com.devonfw.tools/test-project-for-deep-license-scan@0.1.0"))
+        .thenReturn("pkg/maven/com/devonfw/tools/test-project-for-deep-license-scan/0.1.0");
+    DirectUrlWebContentProvider contentProvider = new DirectUrlWebContentProvider(false);
+
+    this.fileScancodeRawComponentInfoProvider = new FileScancodeRawComponentInfoProvider(contentProvider,
+        packageURLHandler);
+    this.fileScancodeRawComponentInfoProvider.setRepoBasePath("src/test/resources/scancodefileadapter/Source/repo");
+    
+    this.singleFileCurationProvider = new SingleFileCurationProvider(packageURLHandler);
+    this.singleFileCurationProvider.setCurationsFileName("src/test/resources/scancodefileadapter/curations.yaml");
+    
+    this.uncuratedScancodeComponentInfoProvider = new UncuratedScancodeComponentInfoProvider(
+        this.fileScancodeRawComponentInfoProvider, packageURLHandler, this.singleFileCurationProvider);
+    this.uncuratedScancodeComponentInfoProvider.setRepoBasePath("src/test/resources/scancodefileadapter/Source/repo");
+
+  }
+  
+  /**
+   * Test the {@link UncuratedScancodeComponentInfoProvider#getComponentInfo(String,String)} method when no paths are excluded
+   * 
+   * @throws ComponentInfoAdapterException if something goes wrong
+   */
+    @Test
+    public void testGetComponentInfoWithoutExclusions() throws ComponentInfoAdapterException {
+    	
+      // given
+      this.singleFileCurationProvider.setCurationsFileName("src/test/resources/scancodefileadapter/nonexisting.yaml");
```

Comment:
This references a nonexisting curations file - so no curation at all (and not just "curation info without exclusions". Is this intended? Maybe we should have three tests:
- no curations existing (actually this is the case here - but not reflected in the name of the test)
- curations extisting without exclusions (the name implies this, but then you should use some existing curations file)
- curations existing with exclusions (see testGetComponentInfoWithCurationsAndExclusions() )

Reply:
The test is copied (and slightly modified) from _class ScancodeComponentInfoAdapterTest {} -testGetComponentInfoWithoutCurations{}_. The curation file not existing is intended. I concede that the naming is confusing. I will add a test for curations file existing without exclusions so that all 3 cases are covered.